### PR TITLE
[addons] Show both enabled and disabled incompatbile addons when performing a migration

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Follow on from https://github.com/xbmc/xbmc/pull/18087

As incompatible addons are now disabled with a reason these would have been missing from the incompatible list when migrating and the user would not be shown which addons would be disabled.

## Motivation and Context

To make it clear to users which addons will be disabled.

## How Has This Been Tested?

Still needs to be tested.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/AddonManager.h [hunk 1]
```diff
@@ -124,14 +124,24 @@ namespace ADDON
      */
     bool FindAddons();
 
+    /*!
+     * @brief Fills the the provided vector with the list of incompatible
+     * enabled addons and returns if there's any.
+     *
+     * @param[out] incompatible List of incompatible addons
```
Comment (enen92) [inline xbmc/addons/AddonManager.h:131]: in,out

File: xbmc/addons/AddonManager.cpp [hunk 2]
```diff
@@ -364,9 +364,17 @@ bool CAddonMgr::GetAddonsInternal(const TYPE& type,
   return addons.size() > 0;
 }
 
-bool CAddonMgr::GetIncompatibleAddons(std::vector<AddonInfoPtr>& incompatible) const
+bool CAddonMgr::GetIncompatibleEnabledAddons(std::vector<AddonInfoPtr>& incompatible) const
+{
+  return GetIncompatibleAddons(incompatible, false);
+}
```
Comment (AlwinEsch) [inline xbmc/addons/AddonManager.cpp:370]: This is OK, only maybe not good to have the `GetIncompatibleAddons` also public then where outside can call this `GetIncompatibleAddons(incompatible, false)` too.

Then maybe better to leave away or make the `GetIncompatibleAddons` private and add another function with
```cpp
bool CAddonMgr::GetIncompatibleAllAddons(std::vector<AddonInfoPtr>& incompatible) const
{
  return GetIncompatibleAddons(incompatible, true);
}
```

File: xbmc/addons/AddonManager.h [hunk 3]
```diff
@@ -124,14 +124,24 @@ namespace ADDON
...
     /*!
      * @brief Fills the the provided vector with the list of incompatible
      * addons and returns if there's any.
      *
      * @param[out] incompatible List of incompatible addons
+     * @param[in] whether or not to include incompatible addons that are disabled
      * @return true if there are incompatible addons
      */
-    bool GetIncompatibleAddons(std::vector<AddonInfoPtr>& incompatible) const;
+    bool GetIncompatibleAddons(std::vector<AddonInfoPtr>& incompatible, bool includeDisabled) const;
```
Comment (AlwinEsch) [inline xbmc/addons/AddonManager.h:144]: Maybe better to rename this and the new then to `...AddonInfos` on end, and to use the `...Addons` only for them wher give the **`CAddon/AddonPtr`**.

So from function name can direct seen which part it give.

**Review by AlwinEsch — COMMENTED** [review]
From first view it looks good and hope to have time for test the next days.

File: xbmc/addons/AddonManager.cpp (hunk 2 above)
Comment (phunkyfish) [reply xbmc/addons/AddonManager.cpp]: Done, went with making it private.

File: xbmc/addons/AddonManager.h (hunk 3 above)
Comment (phunkyfish) [reply xbmc/addons/AddonManager.h]: Done

**Comment by Dnkhatri** [comment]
I know this is off topic but would it be possible to get what addons will work with  v19 before installing it. On android its not possible to install both kodi versions side by side so installing v19 means we cant go back to v18 if some addons are not available.

**Comment by phunkyfish** [comment]
> I know this is off topic but would it be possible to get what addons will work with v19 before installing it. On android its not possible to install both kodi versions side by side so installing v19 means we cant go back to v18 if some addons are not available.

Not even close to on topic 😉
I would recommend installing kodi somewhere else, a laptop or something and seeing if the addons you use are available on kodi 19. If not, contact the author and request a kodi 19 version.

**Comment by phunkyfish** [comment]
@AlwinEsch @enen92 this should be good now. You ok to approve?

**Review by AlwinEsch — APPROVED** [review]
Yes looks good now :+1:

**Review by enen92 — APPROVED** [review]
Tested and works as expected. On the first migration the dialog is shown again with the non-compatible/disabled addons. On subsequent executions the dialog is no longer shown.
Thanks for the quick fix.

**Pull request merged**